.github/workflows/ports_stm32.yml

```diff
@@ -18,20 +18,18 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
-  build_pyb:
+  build_stm32:
+    strategy:
+      fail-fast: false
+      matrix:
+        ci_func:  # names are functions in ci.sh
+          - stm32_pyb_build
+          - stm32_nucleo_build
     runs-on: ubuntu-20.04
     steps:
     - uses: actions/checkout@v4
     - name: Install packages
       run: source tools/ci.sh && ci_stm32_setup
-    - name: Build
-      run: source tools/ci.sh && ci_stm32_pyb_build
+    - name: Build ci_${{matrix.ci_func }}
+      run: source tools/ci.sh && ci_${{ matrix.ci_func }}
```

Comment:
Yeah, this factoring is nice, it eliminates duplication in this YAML file setting up the environment (even though it's pretty simple steps).

Reply:
@dpgeorge Yeah, here it was more a desire to use the same pattern consistently rather than a real win.